src/lib/secrets.js

```diff
@@ -1,33 +1,149 @@
+      port,
+      method: 'POST',
+      headers: {
+        Authorization: `Bearer ${secretToken}`,
+        'Content-Type': 'application/json',
+        Accept: 'application/json',
+        'Content-Length': requestBody ? Buffer.byteLength(requestBody) : 0,
+      },
+    }
+
+    const req = https.request(options, (res) => {
+      if (res.statusCode !== 200) {
+        return reject(new Error(res.statusCode))
       }
+
+      let body = []
+
+      res.on('data', (chunk) => {
+        body.push(chunk)
+      })
+
+      res.on('end', () => {
+        const data = Buffer.concat(body).toString()
+        try {
+          body = JSON.parse(data)
+        } catch (error) {
+          reject(error)
+        }
+        resolve(body)
+      })
+    })
+
+    req.on('error', (error) => {
+      reject(error.message)
     })
-    if (serviceSecrets.length !== 0) {
-      // No Object.fromEntries in node < 12
-      return {
-        ...secrets,
-        [serviceName]: serviceSecrets.reduce((acc, [tokenName, token]) => ({ ...acc, [tokenName]: token }), {}),
+
+    req.write(requestBody)
+
+    req.end()
+  })
+}
+
+const formatSecrets = (result) => {
+  const services =
+    result.data && result.data.me && result.data.me.serviceMetadata && result.data.me.serviceMetadata.loggedInServices
+
+  if (services) {
+    const newSecrets = services.reduce((acc, service) => {
+      const normalized = serviceNormalizeOverrides[service.service] || camelize(service.friendlyServiceName)
+      // eslint-disable-next-line no-param-reassign
+      acc[normalized] = service
+      return acc
+    }, {})
+
+    return newSecrets
+  }
+
+  return {}
+}
+
+// Note: We may want to have configurable "sets" of secrets,
+// e.g. "dev" and "prod"
+const getSecrets = async () => {
+  const secretToken = process.env.ONEGRAPH_AUTHLIFY_TOKEN
+
+  if (!secretToken) {
+    console.warn(
+      'withSecrets is not set up. Visit Netlify Labs to enable it or trigger a new deploy if it has been enabled.',
+    )
+    return {}
+  }
+
+  // We select for more than we typically need here
+  // in order to allow for some metaprogramming for
+  // consumers downstream. Also, the data is typically
+  // static and shouldn't add any measurable overhead.
+  const doc = `query FindLoggedInServicesQuery {
+    me {
+      serviceMetadata {
+        loggedInServices {
+          friendlyServiceName
+          service
+          isLoggedIn
+          bearerToken
+          grantedScopes {
+            scope
+            scopeInfo {
+              category
+              scope
+              display
+              isDefault
+              isRequired
+              description
+              title
+            }
+          }
+        }
       }
     }
-    return secrets
-  }, {})
+  }`
+
+  const body = JSON.stringify({ query: doc })
+
+  // eslint-disable-next-line node/no-unsupported-features/node-builtins
+  const result = await oneGraphRequest(secretToken, new TextEncoder().encode(body))
```

Comment:
`encode()` returns a `Uint8Array`, but `ClientRequest` expects a `Buffer`. Newer versions of Node seem to be forgiving about this, but Node 12 fails. I'd suggest passing `body` to `oneGraphRequest`, and then call `Buffer.from(new TextEncoder().encode(requestBody))` in there.